lib/order.rb

```diff
@@ -0,0 +1,50 @@
+require_relative 'text'
+
+class Order
+  attr_reader :menu, :current, :cost
+
+  def initialize(menu = Takeaway::DISHES, text = Text)
+    @menu = menu
+    @current = Hash.new { |h, k| h[k] = 0 }
+    @cost = 0
+    @text = text
+  end
+
+  def add_to_order(dish, quantity)
+    fail "That isn't on the menu" unless menu.has_key?(dish.to_sym)
+    update_current(dish, quantity)
+    add_to_cost(dish, quantity)
+  end
+
+  def breakdown
+    fail no_items if @cost.zero?
+    order_breakdown
+  end
+
+  def pay(amount)
+    fail no_items if @cost.zero?
```

Comment:
If you want to write a custom error try exploring:

```ruby
def your_method
  fail NoItemsError if cost.zero? # which is a bit bizarre that you look at cost to see if there are items, not... the items
...
end
# outside of the class
class NoItemsError < StandardErrror; end
```

https://ruby-doc.org/core-2.2.3/Exception.html